features/confirmModal/ConfirmModal.tsx

```diff
@@ -0,0 +1,85 @@
+import { CloseIcon } from '@/assets/icons/close';
+import { Button } from '@/shared/ui';
+import { Typography } from '@/shared/ui';
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogOverlay,
+  DialogPortal,
+  DialogTitle,
+} from '@radix-ui/react-dialog';
+
+import style from './ConfirmModal.module.scss';
+
+type ConfirmModalProps = {
+  children: ReactNode;
+  className?: string;
+  openConfirmModal: boolean;
+  openParentModal: boolean;
+  setOpenConfirmModal: (openConfirmModal: boolean) => void;
+  setOpenParentModal: (open: boolean) => void;
+  showCloseButton: boolean;
+  title: string;
+};
+
+export const ConfirmModal = (props: ConfirmModalProps) => {
+  const {
+    children,
+    className,
+    openConfirmModal,
+    openParentModal,
+    setOpenConfirmModal,
+    setOpenParentModal,
+    showCloseButton,
+    title,
+  } = props;
+
+  const closeAllModal = () => {
+    setOpenParentModal(!openParentModal);
+    setOpenConfirmModal(!openConfirmModal);
+  };
+
+  return (
+    <Dialog onOpenChange={closeAllModal} open={openConfirmModal}>
+      {openConfirmModal && (
```

Comment:
This condition is simply unnecessary. The `open` prop controls the display of the contents of the modal window.
Because of this condition, `forceMount` has to be used, which can also be avoided.